src/fheroes2/castle/castle_dialog.cpp

```diff
@@ -457,6 +457,20 @@ Castle::CastleDialogReturnValue Castle::OpenDialog( const bool openConstructionW
             else if ( hero && le.isMouseRightButtonPressedInArea( rectSign2 ) ) {
                 Dialog::QuickInfo( *hero );
             }
+            else if ( hero && HotKeyPressEvent( Game::HotKeyEvent::ARMY_SWAP ) ) {
+                const Troops temp = hero->GetArmy().getTroops();
+
+                hero->GetArmy().Assign( _army );
+                _army.Assign( temp );
+
+                topArmyBar.ResetSelected();
+                bottomArmyBar.ResetSelected();
+
+                topArmyBar.Redraw( display );
+                bottomArmyBar.Redraw( display );
+
+                display.render();
```

Comment:
We can optimize this part a little:

```suggestion
                const fheroes2::Rect & topArea = topArmyBar.GetArea();
                const fheroes2::Rect & bottomArea = bottomArmyBar.GetArea();
                const fheroes2::Rect roi( topArea.x, topArea.y, topArea.width, bottomArea.y + bottomArea.height - topArea.y );

                display.render( roi );
```